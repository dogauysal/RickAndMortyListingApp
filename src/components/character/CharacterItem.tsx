import { AspectRatio, Box, Center, Flex, Heading, Icon, IconButton, Image, Stack, Text } from "native-base";
import React from "react";
import { Character } from "../../models/character/Character";
import CharacterStatus from "./CharacterStatus";
import IonIcons from 'react-native-vector-icons/Ionicons';
import { Dimensions, StyleSheet } from "react-native";


interface IProps {
    item: Character
    onClick?: () => void;
    isDetailScreen: boolean;
}

const CharacterItem: React.FC<IProps> = ({
    item,
    onClick,
    isDetailScreen
}) => {
    return (
        <Box alignItems="center" py={2}>
            <Box width={Dimensions.get('screen').width} rounded="lg" overflow="hidden" px={5} >
                <Box>
                    <AspectRatio w="100%">
                        <Image source={{
                            uri: item.image
                        }} alt="image" />
                    </AspectRatio>
                </Box>
                <Flex py={1} direction="row" justifyContent="space-between">
                    <Stack>
                        <Heading size="sm" color={"gray.500"}>
                            {item.name}
                        </Heading>
                        <CharacterStatus status={item.status} species={item.species} />
                        {isDetailScreen && <Text style={styles.subtitle}>{item.origin.name}</Text>}
                    </Stack>
                    <Stack>
                        {!isDetailScreen ? (
                            <IconButton
                                icon={<Icon as={IonIcons} name="chevron-forward-outline" size={8} color={"black"} />}
                                onPress={onClick}
                            />
                        ) : (
                            <Text style={styles.subtitle}>{`${item.type} | ${item.gender}`}</Text>
                        )}

                    </Stack>
                </Flex>
            </Box>
        </Box>
    )
}

const styles = StyleSheet.create({
    subtitle: {
        fontSize: 12,
        fontStyle: "italic"
    }
})

export default CharacterItem;