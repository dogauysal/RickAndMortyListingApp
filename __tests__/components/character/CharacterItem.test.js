import { NativeBaseProvider } from "native-base";
import React from "react";
import renderer from "react-test-renderer";
import CharacterItem from "../../../src/components/character/CharacterItem";

describe("Character Item Component", () => {
    test("Character Item renders correctly", () => {

        const characterItem = renderer.create(<NativeBaseProvider><CharacterItem /></NativeBaseProvider>).toJSON();
        expect(characterItem).toMatchSnapshot()
    })
})