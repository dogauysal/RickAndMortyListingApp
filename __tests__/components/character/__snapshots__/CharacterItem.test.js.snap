// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Character Item Component Character Item renders correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
